chain/filecoin/client.go

```diff
@@ -48,15 +51,15 @@ func DefaultClientOptions() ClientOptions {
 
 // WithRPCURL returns a modified version of the options with the given API
 // rpc-url
-func (opts ClientOptions) WithRPCURL(rpcURL string) ClientOptions {
-	opts.RPCURL = rpcURL
+func (opts ClientOptions) WithRPCURL(rpcURL pack.String) ClientOptions {
+	opts.RPCURL = string(rpcURL)
 	return opts
 }
 
 // WithAuthToken returns a modified version of the options with the given
 // authentication token.
-func (opts ClientOptions) WithAuthToken(authToken string) ClientOptions {
-	opts.AuthToken = authToken
+func (opts ClientOptions) WithAuthToken(authToken pack.String) ClientOptions {
```

Comment:
It is fine to leave this as-is for now, because it is not too important, but keep in mind that `pack` is a wire format. So unless you are storing/loading this in the context of the function, you should just accept `string`. Obviously, in this context (in fact, most contexts throughout the Multichain) it is not the case that these values are being stored/loaded. For types like `UTXOutput` using `pack` makes sense, because we know that these values are sent/received over a network.